sw/device/silicon_creator/lib/drivers/flash_ctrl.h

```diff
@@ -4,14 +4,158 @@
 #ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_FLASH_CTRL_H_
 #define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_FLASH_CTRL_H_
 
+#include "sw/device/lib/base/bitfield.h"
 #include "sw/device/lib/base/multibits.h"
-#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
 #include "sw/device/silicon_creator/lib/error.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * Flash memory banks.
+ */
+enum {
+  /**
+   * Flash bank 0.
+   */
+  kFlashCtrlBank0 = 0,
+  /**
+   * Flash bank 1.
+   */
+  kFlashCtrlBank1 = 1,
+};
+
+/**
+ * A flash partition.
+ *
+ * Each `flash_ctrl_partition_t` enumeration constant is a bitfield with the
+ * following layout:
+ * - Bit 0: Data (0) or information (1) partition.
+ * - Bits 1-2: Information partition type [0, 2].
+ */
+typedef enum flash_crtl_partition {
+  /**
+   * Data Partition.
+   */
+  kFlashCtrlPartitionData = 0,
+  /**
+   * Information partition of type 0.
+   *
+   * This partition has 10 pages.
+   */
+  kFlashCtrlPartitionInfo0 = (0 << 1) | 1,
```

Comment:
I am not sure if ROM, or ROM_EXT will require access to `kFlashCtrlPartitionInfo1` and `kFlashCtrlPartitionInfo2`.  It would be good to clarify with @tjaychen. 

Once we add more details on the purpose of type1 and type2 we can then add more descriptive names. Let's sync with Tim offline.

Reply:
Captured in #9127.